### PR TITLE
[CS] Apply `SlevomatCodingStandard.Commenting.AnnotationName` rule

### PR DESCRIPTION
**Pull request by phansys**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | n/a

#### Summary

Normalize the usage for the `@inheritdoc` tag.

#### ToDo

- [x] Check if these changes can be covered by PHPCS.

### CODE REVIEW

**Comment by derrabus** [comment]
This looks like something PHPCS should do.

**Comment by phansys** [comment]
I agree, but I don't know if PHPCS has a specific rule for this or if the [`FunctionCommentSniff`](https://github.com/squizlabs/PHP_CodeSniffer/blob/ed8e00df0a83aa96acf703f8c2979ff33341f879/src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php#L17) rule should be enough in this case.

**Comment by greg0ire** [comment]
Can you please open an issue about this on https://github.com/slevomat/coding-standard ?

**Review by derrabus — CHANGES_REQUESTED** [review]
Unless we can have this fixed by PHPCS, I'd like to keep everything as inconsistent as it is.

**Comment by phansys** [comment]
>Can you please open an issue about this on https://github.com/slevomat/coding-standard ?


Seems that both variants are intentionally allowed there. See:
* https://github.com/slevomat/coding-standard/blob/c831e3aa3aeed3e0c4c7cabc3a35652347ea9e50/SlevomatCodingStandard/Sniffs/Commenting/UselessInheritDocCommentSniff.php#L51-L53 (`i` modifier is used for the regex);
* https://github.com/slevomat/coding-standard/blob/c831e3aa3aeed3e0c4c7cabc3a35652347ea9e50/SlevomatCodingStandard/Helpers/DocCommentHelper.php#L120 (`stripos()` is called instead of `strpos()`).

**Comment by greg0ire** [comment]
I believe both are correct and mean the same thing, but I might be wrong. The sniff that detects such comments that are useless should absolutely detect both. The helper should also detect both.
If they both mean the same thing, then it's a matter of style, and a rule to pick one style over the other is 100% the job of a coding standard.

**Comment by phansys** [comment]
>Unless we can have this fixed by PHPCS, I'd like to keep everything as inconsistent as it is.

@derrabus: doctrine/coding-standard#311.

File: composer.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
         "psr/log": "^1|^2|^3"
     },
     "require-dev": {
-        "doctrine/coding-standard": "11.1.0",
+        "doctrine/coding-standard": "12.0.0",
```
Comment (phansys) [inline composer.json]: BTW, if we are releasing new CS rules from "doctrine/coding-standard" only in major versions, couldn't we use `^12.0` here?

Comment (greg0ire) [reply composer.json]: You're right, we probably should… not sure why we are pinning this one.

**Review by greg0ire — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Pull request merged**